code/modules/mob/living/carbon/carbon.dm

```diff
@@ -455,7 +455,7 @@
 		return 0
 	return ..()
 
-/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, toxic = FALSE, harm = TRUE, force = FALSE, purge = FALSE)
+/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, vomit_type = VOMIT_TOXIC, harm = TRUE, force = FALSE, purge_ratio = 0.1)
```

Comment:
This seems like a semi large change

Reply:
The functionality is unchanged, it's a refactor to make it more flexible. By the way, the toxic parameter did nothing at all.